lib/datadog/tracing/contrib/active_support/cache/instrumentation.rb

```diff
@@ -112,7 +108,26 @@ def read(*args, &block)
                 payload = {
                   action: Ext::RESOURCE_CACHE_GET,
                   key: args[0],
-                  tracing_context: {}
+                  tracing_context: {},
+                  # The name of the store is never saved anywhere.
+                  # ActiveSupport looks up stores by converting a symbol into a 'require' path,
+                  # then "camelizing" it for a `const_get` call:
+                  # ```
+                  # require "active_support/cache/#{store}"
+                  # ActiveSupport::Cache.const_get(store.to_s.camelize)
+                  # ```
+                  # @see https://github.com/rails/rails/blob/261975dbef77731d2c76f907f1076c5132ebc0e4/activesupport/lib/active_support/cache.rb#L139-L149
+                  #
+                  # As `self` is the store object, we can reverse engineer
+                  # the original symbol by converting the class name to snake case:
+                  # e.g. ActiveSupport::Cache::RedisStore -> active_support/cache/redis_store
+                  # In this case `redis_store` is the store name.
+                  #
+                  # Because there's no API retrieve only the class name
+                  # (only `RedisStore`, and not `ActiveSupport::Cache::RedisStore`)
+                  # the easiest way to retrieve the store symbol is to convert the fully qualified
+                  # name using Rails-provided methods, then extracting the last part.
+                  store: self.class.name.underscore.match(/active_support\/cache\/(.*)/)[1]
```

Comment:
Would using regex for each command executed on the cache cause performance issues for the customer's application? 

We could memoize the result on an instance variable 

```ruby
def store_name
  return @store_name if defined?(@store_name)
  @store_name = self.class.name.underscore.match(/active_support\/cache\/(.*)/)[1]
  @store_name
end
```

Also, seems we always are getting the last portion of the match. Could we use `split(''/'').last` rather than a regex? 

```ruby
```ruby
def store_name
  return @store_name if defined?(@store_name)
  @store_name = self.class.name.underscore.split('/').last
  @store_name
end
```
```

Reply:
Should we add a comment to explain that `underscore` is from `Rails::ActiveSupport` and that this codes works because is executed within the context of Rails

Reply:
Caching is great, I made those changes.

Reply:
Regarding "split+last" vs regex, I'm keeping the regex because it's the version that more closely mirrors ActiveSupport's logic (`require "active_support/cache/#{store}"`).

And because we are only doing it once, as we are now caching, the performance impact will be almost nil.